batch/batch/worker/worker.py

```diff
@@ -652,6 +696,16 @@ def gcsfuse_path(self, bucket):
     def gsa_key_file_path(self):
         return f'{self.scratch}/gsa-key'
 
+    @classmethod
+    def create(cls, batch_id, user, gsa_key, job_spec, format_version, task_manager):
+        type = job_spec['process']['type']
+        if type == 'docker':
+            return DockerJob(batch_id, user, gsa_key, job_spec, format_version, task_manager)
+        elif type == 'jvm':
+            return JVMJob(batch_id, user, gsa_key, job_spec, format_version, task_manager)
+        else:
+            raise Exception(f"unsupported job type {type}")
```

Comment:
Let's stick to assert's for impossible states. Also, in services code we stick to pylint style, so no-else-return.

```python3
if type == 'docker':
    return DockerJob(...)
assert type == 'jvm'
return JVMJob(...)
```